code/modules/research/nanites/nanite_programs/utility.dm

```diff
@@ -276,3 +276,90 @@
 		if(fault == src)
 			return
 		fault.software_error()
+
+/datum/nanite_program/dermal_button
+	name = "Dermal Button"
+	desc = "Displays a button on the host's skin, which can be used to send a signal to the nanites."
+	extra_settings = list("Sent Code","Button Name","Icon","Color")
+	unique = FALSE
+	var/datum/action/innate/nanite_button/button
+	var/button_name = "Button"
+	var/icon = "power"
+	var/color = "green"
+	var/sent_code = 0
+
+/datum/nanite_program/dermal_button/set_extra_setting(user, setting)
+	if(setting == "Sent Code")
```

Comment:
Can't this and `get_extra_setting()` below be a switch?

Reply:
I suppose, but it would warrant reworking all existing ones along with it, which would be out of scope here.

Reply:
w/e ok